### PR TITLE
[wasm] generate icall_trampoline_dispatch as nested branches

### PR DESCRIPTION
**Pull request by lewing**
Just an experiment to replace the generated list of strcmp calls with nested ifs.

### CODE REVIEW

**Comment by vargaz** [comment]
Note that this function is not called very often in theory.

**Comment by lewing** [comment]
yeah, I don't expect much real impact I just wanted to put this somewhere

File: mono/mini/m2n-gen.cs [hunk 1]
```diff
@@ -174,7 +176,54 @@ class Driver {
...
+	}
+
+	static void WriteDispatch (string [] items)
+	{
+		Array.Sort (items);
+		Console.WriteLine ("static void\nicall_trampoline_dispatch (const char *cookie, void *target_func, InterpMethodArguments *margs)");
+		Console.WriteLine ("{");
+		Partition (items, 0);
+		Console.WriteLine ("\tprintf (\"CANNOT HANDLE COOKIE %s\\n\", cookie);");
+		Console.WriteLine ("\tg_assert (0);");
```
Comment (lewurm) [inline mono/mini/m2n-gen.cs]: This should use `g_error ("...")`. In general, `printf` doesn't work as expected on every platform so we use `g_print` instead. `g_error` will use it accordingly.

**Comment by lewing** [comment]
@monojenkins build OS X x64 Android SDK

**Comment by lewing** [comment]
@monojenkins build OS X x64 Android SDK

**Comment by vargaz** [comment]
Turns out this function is called often, since instead of returning a function pointer, its actually makes the call.

**Pull request merged**